src/registrar/services/dns_host_service.py

```diff
@@ -207,11 +221,38 @@ def save_db_zone(self, vendor_zone_data, domain_name):
             dns_domain = Domain.objects.get(name=domain_name)
 
             dns_zone, _ = DnsZone.objects.get_or_create(dns_account=dns_account, domain=dns_domain, name=zone_name)
-            # Assign ManyToMany field vendor_dns_zone manually because we cannot directly assign forward
-            # side of a many to many set in Django
-            dns_zone.vendor_dns_zone.add(vendor_dns_zone)
 
-            ZonesJoin.objects.get_or_create(
+            ZonesJoin.objects.create(
                 dns_zone=dns_zone,
                 vendor_dns_zone=vendor_dns_zone,
             )
+
+    def save_db_record(self, x_zone_id, vendor_record_data):
+        record_data = vendor_record_data["result"]
+        x_record_id = record_data["id"]
+
+        with transaction.atomic():
+            vendor_dns_record = VendorDnsRecord.objects.create(
+                x_record_id=x_record_id,
+                x_created_at=record_data["created_on"],
+                x_updated_at=record_data["created_on"],
+            )
+
+            # Find record's zone
+            vendor_dns_zone = VendorDnsZone.objects.filter(x_zone_id=x_zone_id).first()
```

Comment:
Is there a reason to do a `.filter().first()` here instead of a `.get`? you'd have to handle a `DoesNotExist` with a get, but maybe we do want to catch that? If vendor_dns_zone doesn't exist, then the next line would fail. Or is it that we are confident enough that it exists that we know we wouldn't hit that error, so using `.first` is cleaner?